ldap/servers/slapd/search.c

```diff
@@ -420,8 +421,16 @@ do_search(Slapi_PBlock *pb)
     if (!psearch || rc != 0 || err != 0) {
         slapi_ch_free_string(&fstr);
         slapi_filter_free(filter, 1);
-        slapi_pblock_get(pb, SLAPI_SEARCH_ATTRS, &attrs);
-        charray_free(attrs);    /* passing NULL is fine */
+
+        /* Get attrs from pblock if it was set there, otherwise use local attrs */
+        char **pblock_attrs = NULL;
+        slapi_pblock_get(pb, SLAPI_SEARCH_ATTRS, &pblock_attrs);
+        if (pblock_attrs != NULL) {
+            charray_free(pblock_attrs); /* Free attrs from pblock */
```

Comment:
I suggest to reset the pblock (slapi_pblock_get(pb, SLAPI_SEARCH_ATTRS, NULL)), to prevent a risk of double free

Reply:
Good point, added.